test/logflare_web/controllers/api/source_controller_test.exs

```diff
@@ -42,4 +45,107 @@ defmodule LogflareWeb.Api.SourceControllerTest do
       assert_lists_equal(response, expected)
     end
   end
+
+  describe "create/2" do
+    test "creates a new source for an authenticated user", %{
+      conn: conn,
+      users: [u1, _u2]
+    } do
+      name = TestUtils.random_string()
+
+      conn
+      |> login_user(u1)
+      |> post("/api/sources", %{name: name})
+      |> response(204)
+
+      assert Repo.one(
```

Comment:
should assert on response instead, controller layer shouldn't have repo knowledge.

Reply:
the idea here was to just create the entity and reply an empty body. I can change it to have the entity in the response body and assert on it.